homeassistant/components/somfy/cover.py

```diff
@@ -0,0 +1,106 @@
+"""
+Support for Somfy Covers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/cover.somfy/
+"""
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION, \
+    ATTR_TILT_POSITION
+from homeassistant.components.somfy import DOMAIN, SomfyEntity, DEVICES, API
+
+DEPENDENCIES = ['somfy']
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Somfy cover platform."""
+    def get_covers():
+        """Retrieve covers."""
+        from pymfy.api.devices.category import Category
+
+        categories = {Category.ROLLER_SHUTTER.value,
+                      Category.INTERIOR_BLIND.value,
+                      Category.EXTERIOR_BLIND.value}
+
+        devices = hass.data[DOMAIN][DEVICES]
+
+        return [SomfyCover(cover, hass.data[DOMAIN][API]) for cover in
+                devices if
+                categories & set(cover.categories)]
+
+    async_add_entities(await hass.async_add_job(get_covers), True)
```

Comment:
`hass.async_add_job` is legacy. Use `hass.async_add_executor_job` if needed.

Reply:
`async_add_entities(await hass.async_add_executor_job(get_covers), True)` will be enough?

Reply:
yes